seed/seed.go

```diff
@@ -73,14 +78,29 @@ func (s *Seeder) Seed(cfg *config.Config) error {
 			}
 			value, err := gtr.Get(key)
 			if err != nil {
-				return err
+				log.Errorf("failed to get consul key %s for field %s: %v", key, f.Name, err)
+				continue
+			}
+			if value == "" {
```

Comment:
That case should be handled differently because it might be a case where empty string is an actual configuration value.
A better approach will be to return an error from `gtr.Get(key)` if [no pair found](https://github.com/taxibeat/harvester/blob/seeding-flow/seed/consul/getter.go#L49) and return value if found no matter if it's empty string or not.

Reply:
I agree with the nil pair.
Can you give me a example of a configuration setting that is valid to be empty?
The problem is that trying to set a empty string into a float will probably generate a error.
I propose to have both checks, nil if the key does not exist and empty if the key has not been set up for any reason. Empty will not seed the field though since it might override previous seed from env var for instance and would render the configuration unusable i am afraid.

Reply:
@mantzas You are right about empty string in a float value, but that will generate an error in a later step upon value set which is already handled.
The case that my proposal makes sense is only in string values where empty string might be valid. 
So, if seed value is a non empty string and consul value is empty string, it should override the previous.

Reply:
@c0nstantx i propose the following then. i will remove the check here but add it in the Set value method of the field. If it is a float, bool it will return a error, if it is a string it will set the field to empty and it is up to the people that set these values to not make an error. Maybe i will add a warning...

Reply:
@mantzas Great. That sound good